index.js

```diff
@@ -3,53 +3,266 @@ const state = {
+  });
+  render(fruitsFilterArray);
+});
+
+select.addEventListener("change", () => {
+  console.log(select.value);
+  let sortedArray = [...state.items];
+  if (select.value === "incrementing") {
+    sortedArray.sort(
+      (item1, item2) => Number(item1.price) - Number(item2.price)
+    );
+  } else if (select.value === "decrementing") {
+    sortedArray.sort(
+      (item1, item2) => Number(item2.price) - Number(item1.price)
+    );
+  }
+  // select.value !== "decrementing" &&
+  // select.value !== "decrementing"
+  // select.value === "default"
+  // else {
+  //   console.log("its default");
+  //   sortedArray = state.items;
+  // }
+  render(sortedArray);
+});
+
+function render(renderItems) {
+  // let renderArray = [];
+  // if (renderItems.length > 0) {
+  //   renderArray = renderItems;
+  // } else {
+  //   renderArray = state.items;
+  // }
+  storeItems.innerHTML = "";
+  renderItems.forEach((item) => {
+    item.quantity = 1;
+    item.image = `assets/icons/${item.id}.svg`;
+    if (vegetables.includes(item.id)) {
+      item.type = "vegetables";
+    } else {
+      item.type = "fruits";
+    }
+
+    addStoreItem(item);
+  });
+}
+function addToCart(item) {
+  if (state.cart.includes(item)) {
+    console.log(state.cart);
+    item.quantity = item.quantity += 1;
+    const li = document.querySelector("#" + `cart-item-${item.id}`);
+    const span = li.querySelector(".quantity");
+    span.innerText = item.quantity;
+    calculateTotal();
+    return;
+  } else {
+    item.quantity = 1;
+    const li = document.createElement("li");
+    li.setAttribute("id", `cart-item-${item.id}`);
+    const img = document.createElement("img");
+    img.setAttribute("class", "cart--item-icon");
+    img.setAttribute("src", item.image);
+    img.setAttribute("alt", item.name);
+    const p = document.createElement("p");
+    p.innerText = item.name;
+    const minusButton = document.createElement("button");
+    minusButton.setAttribute("class", "remove-btn");
+    minusButton.innerText = "-";
+
+    const span = document.createElement("span");
+    span.setAttribute("class", "quantity");
+    span.innerText = item.quantity;
+    const plusButton = document.createElement("button");
+    plusButton.setAttribute("class", "add-btn");
+    plusButton.innerText = "+";
+    li.appendChild(img);
+    li.appendChild(p);
+    li.appendChild(minusButton);
+    li.appendChild(span);
+    li.appendChild(plusButton);
+    cartItems.appendChild(li);
+    state.cart.push(item);
+    console.log(state.cart);
+    plusButton.addEventListener("click", () => {
+      item.quantity += 1;
+      span.innerText = item.quantity;
+      calculateTotal();
+    });
+
+    minusButton.addEventListener("click", () => {
+      item.quantity -= 1;
+      // state.cart.find((item) => {
+      //   if (item.quantity === 0) {
+      //     li.remove();
+      //     state.cart.splice(item);
+      //   }
+      // });
+      if (item.quantity < 1) {
```

Comment:
I like how you have stipulated any quantity below 1 instead of hardcoding in a 0